src/trixi_include.jl

```diff
@@ -0,0 +1,151 @@
+# Note: We can't call the method below `Trixi.include` since that is created automatically
+# inside `module Trixi` to `include` source files and evaluate them within the global scope
+# of `Trixi`. However, users will want to evaluate in the global scope of `Main` or something
+# similar to manage dependencies on their own.
+"""
+    trixi_include([mod::Module=Main,] elixir::AbstractString; kwargs...)
+
+`include` the file `elixir` and evaluate its content in the global scope of module `mod`.
+You can override specific assignments in `elixir` by supplying keyword arguments.
+It's basic purpose is to make it easier to modify some parameters while running Trixi.jl from the
+REPL. Additionally, this is used in tests to reduce the computational burden for CI while still
+providing examples with sensible default values for users.
+
+Before replacing assignments in `elixir`, the keyword argument `maxiters` is inserted
+into calls to `solve` with it's default value used in the SciML ecosystem
+for ODEs, see the "Miscellaneous" section of the
+[documentation](https://docs.sciml.ai/DiffEqDocs/stable/basics/common_solver_opts/).
+
+# Examples
+
+```jldoctest
+julia> redirect_stdout(devnull) do
+         trixi_include(@__MODULE__, joinpath(examples_dir(), "tree_1d_dgsem", "elixir_advection_extended.jl"),
+                       tspan=(0.0, 0.1))
+         sol.t[end]
+       end
+[ Info: You just called `trixi_include`. Julia may now compile the code, please be patient.
+0.1
+```
+"""
+function trixi_include(mod::Module, elixir::AbstractString; kwargs...)
+    # Check that all kwargs exist as assignments
+    code = read(elixir, String)
+    expr = Meta.parse("begin \n$code \nend")
+    expr = insert_maxiters(expr)
+
+    for (key, val) in kwargs
+        # This will throw an error when `key` is not found
+        find_assignment(expr, key)
+    end
+
+    # Print information on potential wait time only in non-parallel case
+    # if !mpi_isparallel()
+    #     @info "You just called `trixi_include`. Julia may now compile the code, please be patient."
+    # end
```

Comment:
What do we do with this? Do we want to bring MPI stuff here? Or do we omit the message?

Reply:
Argh! Can we just omit it here and in Trixi.jl export our own trixi_include that does the MPI check and then calls TrixiBase.trixi_include?

Reply:
But then we have conflicting exports. I'm not a fan of this message, anyway. `Base.include` doesn't do that either.

Reply:
We added this as a user convenience thing. I liked it, especially since sometimes you get long wait times of 30+ seconds compilation before any output appears. Still, to me it would be OK to remove again. Alternatively, we could make MPI a dependency but that seems overly error prone. What are your thoughts @ranocha ?

Reply:
The Aachen group asked for it, so we included it. I don't have a strong opinion but it may be strange to remove it again. We could do as @sloede suggested?

Reply:
I think the problem that @efaulhaber refers to is in that case we will have conflicts if we load, e.g., Trixi.jl and TrixiShallowWater.jl at the same time, where TrixiSW will export TrixiBase.trixi_include and Trixi will export Trixi.trixi_include. If both export the same TrixiBase.trixi_include, this shouldn't be a problem (or is this already going to cause issues?).

Reply:
Yes, that's correct. There is no good solution. Everything comes with some drawbacks.

Reply:
We _could_ also make a package extension based on whether MPI.jl has been loaded. But that sounds like an awful lot of work for just a "convenience" feature.

The only reasonable way to keep the current functionality seems to be to have MPI as a dependency in TrixiBase (not optimal, but maybe ok), and then do
```julia
if !MPI.Is_initialized() || MPI.Comm_size(MPI_COMM_WORLD) == 1
    @info "..."
end
```
upon each call to `trixi_include`. This would at least preclude us from having to initialize MPI in TrixiBase, which I think is a bad idea.

Anyways, let's move forward with this. Not showing this convenience warning is not a breaking change and while it was nice for usability, it was also not very Julian. So please go ahead @efaulhaber and implement this package without it.